### PR TITLE
[WIP] feat(wandb): log both training and validation data

### PR DESCRIPTION
**Pull request by borisdayma**
Training data and parameters is logged at each batch and validation data at the end of each epoch

### CODE REVIEW

**Comment by borisdayma** [comment]
This let WandbCallback log both training and validation data, as well as prediction samples:

* training data (loss, learning rate…) is logged at each training batch
* validation data and prediction samples are logged at the end of each epoch

Here is [an example W&B run from lesson1-pets](https://app.wandb.ai/borisd13/fastai2/runs/7mls37kx?workspace=user-borisd13).

Few notes:

1. the entire section `after_fit` does not seem needed anymore, please confirm I can remove it - *Note: this type of graphs would actually be better logged with [summary metrics](https://docs.wandb.com/library/python/log#summary-metrics)*
2. `entity` is not needed in `wandb.init` and will automatically use the correct default, please confirm I can remove it
3. I was able to add the WandbCallback in the `fit_one_cycle` method but not directly in the learner, which would be better when doing "multi-phases training" to keep track of accumulated epochs

For item 3. above, I expected to be able to use:
`learn = cnn_learner(dbunch, resnet34, metrics=error_rate, cbs=WandbCallback()).to_fp16()` but got following error: [error stack trace link](https://pastebin.com/e4rqKqPT)

**Comment by sgugger** [comment]
Will try and iter from this. I don't think the `after_fit` section will be necessary anymore, which is also the thing that was making 3 fail, so removing it should hopefully solve that issue. Not setting `entity` did not work for me, at least for the first try. Maybe it's cached somehow?

**Comment by sgugger** [comment]
Just did it on a fresh VM and can confirm that on the first try (in a jupyter notebook) you get an error if you don't specify entity=your_user_name. If it has been done once, it works afterward without passing anything.

**Comment by sgugger** [comment]
Tested more and for some reason, this new version doesn't log anything online. I'll try to find the cause tomorrow.

**Comment by borisdayma** [comment]
That is strange. Do you have any error code you can share?
I'll try it again on a fresh install on my side and will send you my notebook.

**Comment by borisdayma** [comment]
So I tried from a fresh install and there seems to be an issue with wandb CLI. After login, when we do `wandb status`, I believe the entity should be set which is not the case. I'm going to investigate.

In the meantime, you can either pass your entity (below your name on your profile page) or you can pass `wandb.init(anonymous="allow")`. You should not even need a project name (it's mainly to keep all your runs organized).

Please note that if you have an error at `wandb.init`, the best is to restart the kernel.

You can see here my notebook.
[lesson1-pets-wandb.zip](https://github.com/fastai/fastai_dev/files/3812700/lesson1-pets-wandb.zip)

Let me know if there is still any issue.

**Comment by borisdayma** [comment]
I tested again today on a fresh install and all now seems to work fine with `wandb.init()` (without setting "anonymous" or "entity"). It may have been a temporary issue from the server.

The way to check is to type `wandb status` in the shell and check that "entity" is set correctly after you did `wandb login`
